src/resolvers/deck/createDeck.ts

```diff
@@ -28,3 +39,193 @@ export const createDeck = mutationWithClientMutationId({
     return { deck }
   },
 })
+
+export const importDeck = mutationWithClientMutationId({
+  name: 'ImportDeck',
+  description: 'Import/Duplicate a deck entity',
+  inputFields: {
+    id: { type: GraphQLNonNull(GraphQLID), description: 'Public deck id' },
+  },
+  outputFields: {
+    deck: { type: DeckType, description: 'Imported deck' },
+  },
+  mutateAndGetPayload: async (
+    { id },
+    { user, publishedDeckLoader, modelLoader, fieldsByModelLoader }: Context
+  ) => {
+    const helperMap: {
+      updateModelIndex: Record<string, string>
+      updateFieldIndex: Record<string, string>
+      modelTemplates: Record<string, TemplateDocument[]>
+    } = {
+      updateModelIndex: {},
+      updateFieldIndex: {},
+      modelTemplates: {},
+    }
+    const { id: deckId } = fromGlobalId(id)
+    const publishedDeck = await publishedDeckLoader.load(deckId)
+
+    if (!publishedDeck) {
+      return { deck: null }
+    }
+
+    const deckNotes = await NoteModel.find({ deckId: publishedDeck?._id })
+
+    // Find requires models/fields & templates
+    const modelIds = deckNotes.map((note) => note.modelId.toString())
+    const uniqueModelIds = modelIds.filter((v, i) => modelIds.indexOf(v) === i)
+
+    const models = await Promise.all(
+      uniqueModelIds.map(async (modelId) => {
+        const model = await modelLoader.load(modelId)
+        const currentModelFields = await fieldsByModelLoader.load(model?._id)
+        const currentModelTemplates = await TemplateModel.find({
+          modelId: model?._id,
+        })
+
+        return {
+          model,
+          fields: currentModelFields,
+          templates: currentModelTemplates,
+        }
+      })
+    )
+
+    // Create deck
+    const newDeck = await DeckModel.create({
+      title: publishedDeck?.title,
+      description: publishedDeck?.description,
+      ownerId: user?._id,
+      slug: '',
+      published: false,
+      configuration: defaultDeckConfig,
+    })
```

Comment:
something I thought of yesterday, wouldn't it be nice for us to keep the original deck id in the new deck? So we can keep measures like how many people installed a particular deck on the marketplace, etc